### ISSUE TITLE
Stuck with a previous max limit

### ISSUE THREAD

**Issue by Philmod** [tj/node-ratelimiter#10]
Hey guys,

I've started using your module, but I realized that if you change the max limit (after some requests), the module stays limited with the first one.

Thanks,
Philmod

**Comment by noamshemesh** [comment]
Hi Philmod,
How do you change the max limit?
If you're initializing new Limiter that should not be an issue.

Feel free to paste some code here to make your point

Noam

**Comment by Philmod** [comment]
Hey,

In my middleware, I get the rate limits from the database: 
```
var limit = new Limiter({ 
          id: id
        , db: server.redis_ratelimiter
        , max: rateLimiting.max
        , duration: rateLimiting.duration
    });
```
But even if this `max` is updated, the first limit is hit.

**Comment by noamshemesh** [comment]
I'm still not sure what you're trying to do, but if you want to change the maximum limit *after* initialization, you should create a new Limiter. 
The maximum value is set in the constructor and cannot be changed.

**Comment by Philmod** [comment]
Oh ok, so changing the value in the database is not enough to modify the limit?

**Comment by noamshemesh** [comment]
you mean in redis? It works for the same expiration period.

**Comment by Philmod** [comment]
Sorry, I'm not really clear. Let me try again :-)

The `rateLimiting.max` and `rateLimiting.duration` are set in the database.
Let's say the middleware is called with these first values.
Then I changed these `max` and/or `duration` in the database, if that middleware is called again (with the same `id` but different `max`/`duration`), the user will be rate-limited with the first values.

**Comment by noamshemesh** [comment]
ohh, I think I understand.

Sorry if it's too basic and I'm missing the point here, but take a look at this one:
http://stackoverflow.com/questions/518000/is-javascript-a-pass-by-reference-or-pass-by-value-language

To make long story short, when you pass a simple argument (string, integer, etc..) to a function, the value is copied and you can't change it later from outside.

**Comment by Philmod** [comment]
I think the issue is because I use a same `id` but with different `max` value. And redis has the first `max` value in memory?

If I use a different `id` with the new `max`, it will fix it?

**Comment by noamshemesh** [comment]
So, if I hear you correctly, this is the scenario (not tested):
```javascript
var limiter = new Limiter({id: 2, max: 10})
limiter.get(function (err, limit) { 
// Should return 9 
});
limiter = new Limiter({id: 2, max: 20})
limiter.get(function (err, limit) {
// Should return 18(or 19?) but returns 8
});
```

?

**Comment by Philmod** [comment]
Yes :-)

**Comment by noamshemesh** [comment]
a different id will definitely solve the issue, you can append the max value to the id if you want a simple workaround.

I don't think that we should fix this issue, as it's unusual to change the maximum after the value was created in redis. besides that, there is a simple workaround.

Thanks for reporting this one!

**Comment by Philmod** [comment]
Ok, I'm going to create the `id` by hashing the user id + max + duration.

Thanks.
